sql/item_func.cc

```diff
@@ -6330,17 +6330,20 @@ Item_func_sp::fix_fields(THD *thd, Item **ref)
 
   if (m_sp->agg_type() == GROUP_AGGREGATE)
   {
-    List<Item> list;
-    list.empty();
-    for (uint i=0; i < arg_count; i++)
-      list.push_back(*(args+i));
-
     Item_sum_sp *item_sp;
     Query_arena *arena, backup;
     arena= thd->activate_stmt_arena_if_needed(&backup);
 
     if (arg_count)
+    {
+      List<Item> list;
+      list.empty();
+      for (uint i=0; i < arg_count; i++)
+        list.push_back(*(args+i));
```

Comment:
Is it sophisticated way of writing `args[i]`?
Also it'd be nice to add space here: `i= 0`